spec/controllers/sessions_controller_spec.rb

```diff
@@ -27,130 +27,164 @@
   end
 
   describe "GET google" do
-    it "redirects to github" do
+    it "redirects to google" do
       expect(get(:google)).to redirect_to "/auth/google_oauth2"
     end
   end
 
   describe "GET create" do
-    before do
-      request.env["omniauth.auth"] = OmniAuth.config.mock_auth[:github]
-    end
-
-    subject { get :create, params: { provider: "github" } }
-
-    describe "for a new user" do
-      it "redirects to email confirmation step" do
-        expect(subject).to redirect_to get_email_path
-      end
-
-      it "sets omniauth info in cookies" do
-        subject
-        expect(session["uid"]).to eq("12345")
-        expect(session["username"]).to eq("someone")
-        expect(session["provider"]).to eq("github")
+    describe "with Github authentication" do
```

Comment:
I don't think Github generated a very helpful diff for this change so I'll try to explain it to make things clearer.

Here through line 160, I just indented the existing test cases into a `describe "with Github authentication" do ... end` block, and added one assertion and some comments to the `"sets omniauth info in cookies"` test case. The diff makes it look like I deleted / rearranged the test cases or something, but I did not.

Reply:
Thanks for leaving the explanation comments, that's very helpful!

Seems like turning on the "hide whitespace" setting gives a much better diff:
![Screen Shot 2022-07-31 at 2 43 33 PM](https://user-images.githubusercontent.com/6729309/182046572-6d289835-2c15-48c0-8ec1-69a00806483d.png)

![Screen Shot 2022-07-31 at 2 44 09 PM](https://user-images.githubusercontent.com/6729309/182046574-b3b714ce-bbc1-4612-8561-dd87c9a665e5.png)

Reply:
ah thank you! Haven't used Github in a while, still getting used to it again.